### PR TITLE
Add mediafile.move action + tests.

### PR DESCRIPTION
**Pull request by reiterl**
Based on the agenda_item.assign action.
Added a check: parent_id should point to a is_directory mediafile.

### CODE REVIEW

File: openslides_backend/action/mediafile/move.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    def prepare_assign_data(
+        self, parent_id: Optional[int], ids: List[int], meeting_id: int
+    ) -> ActionPayload:
+        filter = FilterOperator("meeting_id", "=", meeting_id)
+        db_instances = self.datastore.filter(
+            collection=self.model.collection,
+            filter=filter,
+            mapped_fields=["id"],
+            lock_result=True,
+        )
```
Comment (jsangmeister) [inline openslides_backend/action/mediafile/move.py:59]: Why do you fetch all instances here? A get_many on the given ids should suffice, I think.

File: openslides_backend/action/mediafile/move.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    def get_updated_instances(self, payload: ActionPayload) -> ActionPayload:
+        # Payload is an iterable with exactly one item
+        instance = next(iter(payload))
+        yield from self.prepare_assign_data(
+            parent_id=instance["parent_id"],
+            ids=instance["ids"],
+            meeting_id=instance["meeting_id"],
+        )
+
+    def prepare_assign_data(
```
Comment (jsangmeister) [inline openslides_backend/action/mediafile/move.py:50]: Please rename.

File: openslides_backend/action/mediafile/move.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            filter=filter,
+            mapped_fields=["id"],
+            lock_result=True,
+        )
+
+        if parent_id is None:
+            for id_ in ids:
+                if id_ not in db_instances:
+                    raise ActionException(f"Id {id_} not in db_instances.")
+                yield {"id": id_, "parent_id": None}
```
Comment (jsangmeister) [inline openslides_backend/action/mediafile/move.py:65]: This is the same as the `else` case, you can just move this behind the if-else

File: openslides_backend/action/mediafile/move.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                    FullQualifiedId(self.model.collection, gp_parent_id), ["parent_id"]
+                )
+            for id_ in ids:
+                if id_ in ancesters:
+                    raise ActionException(
+                        f"Moving item {id_} to one of its children is not possible."
+                    )
+                if id_ not in db_instances:
+                    raise ActionException(f"Id {id_} not in db_instances.")
+                yield {"id": id_, "parent_id": parent_id}
```
Comment (jsangmeister) [inline openslides_backend/action/mediafile/move.py:87]: You may also have to update the `inherited_access_group_ids` as well as `is_public`, depending ob the `access_group_ids` from children and parent. Also add tests for this.

File: tests/system/action/mediafile/test_move.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+                        {"meeting_id": 222, "ids": [7], "parent_id": 8},
+                    ],
+                }
+            ],
+        )
+        self.assert_status_code(response, 400)
+        mediafile_7 = self.get_model("mediafile/7")
+        assert mediafile_7.get("parent_id") is None
+        mediafile_8 = self.get_model("mediafile/8")
+        assert mediafile_8.get("parent_id") is None
```
Comment (jsangmeister) [inline tests/system/action/mediafile/test_move.py:172]: Please add a test where a cycle is formed to make sure it is prevented.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

**Comment by normanjaeckel** [comment]
You also have to rebase because of dependency changes.

File: openslides_backend/action/mediafile/move.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/action/mediafile/move.py]: Okay, did a get many request then.

File: openslides_backend/action/mediafile/move.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/action/mediafile/move.py]: Moved it.

File: openslides_backend/action/mediafile/move.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/mediafile/move.py]: Renamed into prepare_move_data.

File: openslides_backend/action/mediafile/move.py (hunk 4 above)
Comment (reiterl) [reply openslides_backend/action/mediafile/move.py]: Calculated is_public and inherited_access_group_id for item and children, used the mediafile update code for this.

Comment (reiterl) [reply openslides_backend/action/mediafile/move.py]: Also add and update tests.

File: openslides_backend/action/mediafile/calculate_mixins.py [hunk 6]
```diff
@@ -31,3 +32,45 @@ def calculate_inherited_groups(
...
+            for child_id in mediafile["child_ids"]:
+                child = self.datastore.get(
+                    FullQualifiedId(Collection("mediafile"), child_id),
+                    [
+                        "access_group_ids",
+                        "child_ids",
+                        "is_public",
+                        "inherited_access_group_ids",
+                    ],
+                )
```
Comment (jsangmeister) [inline openslides_backend/action/mediafile/calculate_mixins.py:55]: Also execute a `get_many` request here

**Review by jsangmeister — CHANGES_REQUESTED** [review]
rest looks good. After that, you can squash and merge.

File: openslides_backend/action/mediafile/calculate_mixins.py (hunk 6 above)
Comment (reiterl) [reply openslides_backend/action/mediafile/calculate_mixins.py]: Done.

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**